ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/home/view/HomeView.kt

```diff
@@ -189,33 +209,160 @@ private fun HomeMainContent(
                 )
             }
         }
-        Status(walletSnapshot)
+
+        Status(walletSnapshot, updateAvailable)
+
+        Spacer(modifier = Modifier.height(24.dp))
+
         PrimaryButton(onClick = goSend, text = stringResource(R.string.home_button_send))
+
         TertiaryButton(onClick = goRequest, text = stringResource(R.string.home_button_request))
+
         History(transactionHistory)
     }
 }
 
 @Composable
-private fun Status(walletSnapshot: WalletSnapshot) {
-    Column(Modifier.fillMaxWidth()) {
-        Header(text = walletSnapshot.totalBalance().toZecString())
-        Body(
-            text = stringResource(
-                id = R.string.home_status_shielding_format,
-                walletSnapshot.saplingBalance.total.toZecString()
+@Suppress("LongMethod", "MagicNumber")
+private fun Status(walletSnapshot: WalletSnapshot, updateAvailable: Boolean) {
+    // TODO this will go away before PR merged
+    Twig.info { "WALLET: $walletSnapshot" }
+
+    val configuration = LocalConfiguration.current
+    val contentSizeRatioRatio = if (configuration.orientation == Configuration.ORIENTATION_LANDSCAPE) {
+        0.45f
+    } else {
+        0.9f
+    }
+
+    // parts sizes
+    val progressCircleStroke = 12.dp
+    val progressCirclePadding = progressCircleStroke + 6.dp
+    val contentPadding = progressCircleStroke + progressCirclePadding + 10.dp
+
+    // parts values
+    var progressCirclePercentage = 0
+    val zecAmountText = walletSnapshot.totalBalance().toZecString()
+    var usdAmountText = walletSnapshot.spendableBalance().toUsdString()
+    var statusText = ""
+
+    // Note: these reactions on the STATUS need to be enhanced, we provide just an elementary reactions for now.
+    when (walletSnapshot.status) {
```

Comment:
Would it make sense to refactor this `when` to a separate method?  I'm thinking of a pattern similar to `ParseResult` during the Restore UI flow, which allows for unit testing of the state transformation to a new object consumed by the UI.  

This might end up being a bad idea—I'm just thinking out loud on this one since there's a bit of complexity here and a lot of different states.

I was thinking something along the lines of:

`internal fun (walletSnapshot.status): WalletDisplayValues`